### ISSUE TITLE
Raw block has an extra space when text ends with backtick

### ISSUE THREAD

**Issue by Enter-tainer** [typst/typst#3820]
### Description

````typ
  ```typ
  code`
  ```
````

The AST of code above is 
```
Markup: 27 [
    Space: "  ",
    Raw: 23 [
        RawDelim: "```",
        RawLang: "typ",
        RawTrimmed: "\r\n ",
        Text: " code`",
        RawTrimmed: "\r\n  ",
        RawDelim: "```",
    ],
    Space: "\r\n",
]
```

Note there is an extra space in ```Text: " code`",```.

This is because the inner string ends with backtick, then the space in last line is eaten. Then the dedent becomes wrong.
This is discovered when debugging a raw block formatting issue in https://github.com/Enter-tainer/typstyle

This is reproducible in 0.10.0 and 0.11.0

### Operating system

Web app, Windows, Linux, macOS

### Typst version

- [X] I am using the latest version of Typst

**Comment by Leedehai** [comment]
````typ
#show raw: it => { box(fill: green, it) }

  ```typ
  B
  ```

  ```typ
  D`
  ```
````

Regardless of the extra backtick's presence, it appears the extra space always exists:
```
lexer.rs ["", "  B", "  "]
lexer.rs ["", "  D`", " "]
```

However, in yet another suprising twist, the image below does indicate only the latter (i.e. with the extra backtick)'s extra space resulted in a wider bounding box:
<img width="207" src="https://github.com/typst/typst/assets/18319900/c8131659-e00c-468d-aabf-9264241cae6e">

**Comment by Enter-tainer** [comment]
FYI, @Myriad-Dreamin has a fix for this https://github.com/Myriad-Dreamin/typst/commit/d2afbb9a62436a20b507e833d70a2ddc77a00b74

**Comment by Leedehai** [comment]
````typ
#show raw: it => { box(fill: green, it) }

  ```typ
  B
  ```

  ```typ
  D`
  ```
````

Regardless of the extra backtick's presence, it appears the extra space always exists, by printing `lines` at https://github.com/typst/typst/blob/b3615ccd493c4791c3c7565fc7417cba50e8fd71/crates/typst-syntax/src/lexer.rs#L309
```
lexer.rs ["", "  B", "  "]
lexer.rs ["", "  D`", " "]
```

However, in yet another suprising twist, the image below does indicate only the latter (i.e. with the extra backtick)'s extra space resulted in a wider bounding box:
<img width="207" src="https://github.com/typst/typst/assets/18319900/c8131659-e00c-468d-aabf-9264241cae6e">

**Comment by Enter-tainer** [comment]
> Regardless of the extra backtick's presence, it appears the extra space always exists, by printing lines at

It's not the final product. You may print the ast to see it

**Comment by laurmaedje** [comment]
> FYI, @Myriad-Dreamin has a fix for this Myriad-Dreamin@d2afbb9

@Myriad-Dreamin If you want to open a PR for it, that'd be great!
